Cranelift: bnot.b1 is illegal on x86

While looking into another issue I minimized the following testcase, which produces invalid IR:

```
target x86_64

function u0:323() system_v {
block0:
    v221 = bconst.b1 false
    v222 = bconst.b1 false
    v223 = bnot v221
    v224 = band v223, v222
    trap unreachable
}
```

```
[-]                                 v223 = bnot v221
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
; error: inst2: v223 is a real GPR value defined by a ghost instruction
```

It seems to me that it should work but I don't understand what's the missing legalization or selection rule is in this case. @iximeow?

Tried that, it breaks with a really strange message:
```
error[E0425]: cannot find value `typeof_x` in this scope
   --> /home/whitequark/Projects/wasmtime/target/debug/build/cranelift-codegen-ca1b406eb4035fa6/out/legalizer.rs:195:68
    |
195 |                 let predicate = predicate && TYPE_SETS[0].contains(typeof_x);
    |                                                                    ^^^^^^^^ help: a local variable with a similar name exists: `typeof_a`

error[E0425]: cannot find value `typeof_x` in this scope
   --> /home/whitequark/Projects/wasmtime/target/debug/build/cranelift-codegen-ca1b406eb4035fa6/out/legalizer.rs:197:46
    |
197 |                     let y = pos.ins().iconst(typeof_x, -1);
    |                                              ^^^^^^^^ help: a local variable with a similar name exists: `typeof_a`

error[E0425]: cannot find value `x` in this scope
   --> /home/whitequark/Projects/wasmtime/target/debug/build/cranelift-codegen-ca1b406eb4035fa6/out/legalizer.rs:198:61
    |
198 |                     let a = pos.func.dfg.replace(inst).bxor(x, y);
    |                                                             ^ help: a local variable with a similar name exists: `a`
```

--- a/cranelift/codegen/meta/src/isa/x86/encodings.rs
+++ b/cranelift/codegen/meta/src/isa/x86/encodings.rs
@@ -1454,6 +1454,7 @@ fn define_alu(
     // x86 has a bitwise not instruction NOT.
     e.enc_i32_i64(bnot, rec_ur.opcodes(&NOT).rrr(2));
     e.enc_b32_b64(bnot, rec_ur.opcodes(&NOT).rrr(2));
+    e.enc_both(bnot.bind(B1), rec_ur.opcodes(&NOT).rrr(2));
 
     // Also add a `b1` encodings for the logic instructions.
     // TODO: Should this be done with 8-bit instructions? It would improve partial register

--- a/cranelift/filetests/filetests/isa/x86/bnot-b1.clif
+++ b/cranelift/filetests/filetests/isa/x86/bnot-b1.clif
@@ -1,0 +1,14 @@
+test binemit
+test run
+
+target x86_64
+
+function u0:323() -> b1 {
+block0:
+    [-,%rax] v221 = bconst.b1 false   ; bin: 40 b8 00000000
+    [-,%rcx] v222 = bconst.b1 true    ; bin: 40 b9 00000001
+    [-,%rax] v223 = bnot v221         ; bin: 40 f7 d0
+    [-,%rax] v224 = band v223, v222   ; bin: 40 21 c8
+    return v224
+}
+; run

